pkg/model/deployment_source.proto

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package model;
+option go_package = "github.com/pipe-cd/pipecd/pkg/model";
+
+message DeploymentSource {
+  // The application directory where the source code is located.
+  string application_directory = 1;
+  // The git commit revision of the source code.
+  string revision = 2;
+  // The configuration of the application which is independent for plugins.
+  GenericApplicationSpec generic_application_config = 3;
+  // The configuration of the application which is specific for plugins.
+  PluginApplicationSpec application_config = 4; 
```

Comment:
For these two, I think we need the data structure under the `pkg/config` package. So, how about passing this as bytes to the plugin, and the plugin will import the pipecd `pkg/config` package to unmarshal it? In that case, we can reuse the logic of the marshal/unmarshal config we currently have.

Reply:
I mean, reuse this whole: https://github.com/pipe-cd/pipecd/blob/master/pkg/config/application.go#L29-L61

Reply:
I removed GenericApplicationSpec.
PluginApplicationSpec's fields are api_version, kind, and spec.
The api_version and kind are common for all plugins, so I defined as fields.
The field spec is defined as bytes, as you pointed out.